[discover.swiss] Add spider for 3843 hotels and 54 serviced apartments

brawer opened the pull request:
disover.swiss is the IT backend platform of the Swiss hotels association.

iandees commented:
This looks like an aggregation. Can we add spiders for the individual chains instead?

iandees commented:
This is just a question, by the way. I'm not rejecting this change, just curious what the level of effort is.

brawer commented:
The majority are small hotels, not chains.

brawer commented:
Agree. Actually, we're already tagging the output of some spiders (when writing out GeoJSON) with dataset attributes for licensing, attribution strings, or whether there's permission to bulk import the dataset into OSM. How about using this existing mechanism to tag datasets as aggregates? Then, clients who care could filter them out, or use this as a signal for conflation.

If you like the proposal, what tags should be used?

Cj-Malone commented:
I would like it to be automatic, not something for us to forget to add to dataset_attributes in spider, it can still be there in the output but automatically applied. I would vote to do this based on spider locations. Something like: `locations/spiders` for brand spiders, `locations/government_spiders`, `locations/aggregator_spiders`

This is in the plumbing, I'm happy to do it in another PR.

Cj-Malone commented on locations/spiders/discover_swiss.py:
```diff
@@ -0,0 +1,122 @@
+        # between Hotellerie Suisse (who runs the discover.swiss platform)
+        # and the Swiss OpenStreetMap association.
+        # https://osmfoundation.org/wiki/Licence/Waiver_and_Permission_Templates
+        "attribution": "mandatory",
```
```suggestion
        "attribution": "required",
```

To match the others.

brawer commented:
Have another look? I’ve added dataset attributes, indicating via `aggregate=yes` that the dataset is an aggregate. Actually it’s a mix of unique content (most small hotels aren’t part of any chain) and chains, so perhaps `aggregate=partial` would be more accurate, but I didn’t want to make the tagging overly complicated. I’ve also  improved the tagging of hostels, to handle 320 cases where feed doesn’t comply with its own specification. Finally, I’ve added dataset attributes to indicate the license and permission use in OpenStreetMap; an eventual import to OSM is why I’m writing this scraper in the first place.

Cj-Malone commented:
Sorry, I've been really busy. I'll take a look tomorrow.

> eventual import to OSM

thank you, I feel like so much of this data is wasted

iandees commented:
Yea, these dataset attributes make sense. I think we might want to define a better schema than flat key/value at some point, but this is a great start.

brawer commented:
OK to merge?

Cj-Malone commented on locations/spiders/discover_swiss.py:
```diff
@@ -0,0 +1,122 @@
+        # via standard template. Negotiations took place in January 2025
+        # between Hotellerie Suisse (who runs the discover.swiss platform)
+        # and the Swiss OpenStreetMap association.
+        # https://osmfoundation.org/wiki/Licence/Waiver_and_Permission_Templates
```
If not, please do a follow up PR. I've left you waiting on this PR for too long so I'll merge now. Thank you.
